railgun/controllers/configuration/secret_utils.go

```diff
@@ -0,0 +1,161 @@
+package configuration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/kong/kubernetes-ingress-controller/railgun/controllers"
+	"github.com/kong/kubernetes-ingress-controller/railgun/pkg/configsecret"
+)
+
+// -----------------------------------------------------------------------------
+// Secret Utils - Storage
+// -----------------------------------------------------------------------------
+
+// storeIngressObj reconciles storing the YAML contents of Ingress resources (which are managed by Kong)
+// from multiple versions which remain supported.
+func storeIngressObj(ctx context.Context, c client.Client, log logr.Logger, nsn types.NamespacedName, obj client.Object) (ctrl.Result, error) {
+	// TODO need EVENTS here
+	// TODO need more status updates
+	// TODO: (shane) I want to refactor this into several smaller functions
```

Comment:
```suggestion
	// TODO: (shane) I want to refactor this into several smaller functions
	// TODO: collapse nsn + obj, this is redudant as obj includes nsn
	// TODO: pass in secret namespace as part of function sig instead of env
```